extensions/amazon-lambda/common-deployment/src/main/java/io/quarkus/amazon/lambda/deployment/LambdaUtil.java

```diff
@@ -50,6 +53,17 @@ public static void writeFile(OutputTargetBuildItem target, String name, String o
         Files.write(artifact, output.getBytes(StandardCharsets.UTF_8));
     }
 
+    public static void writeExecutableFile(OutputTargetBuildItem target, String name, String output) throws IOException {
+        writeFile(target, name, output);
+
+        Path artifact = target.getOutputDirectory().resolve(name);
+        Set<PosixFilePermission> permissions = new HashSet<>();
+        permissions.add(PosixFilePermission.OWNER_READ);
+        permissions.add(PosixFilePermission.OWNER_WRITE);
+        permissions.add(PosixFilePermission.OWNER_EXECUTE);
+        Files.setPosixFilePermissions(artifact, permissions);
```

Comment:
Given the failure on Windows, I think what you will have to do is conditionally set these PosixFilePermissions, something like:

```
if (Files.getFileAttributeView(artifact, PosixFilePermission.class) != null) {
... // setup the permissions
}
```

Reply:
According with documentation, I could do just this. Tell me what do you think. 

```
        Path artifact = target.getOutputDirectory().resolve(name);
        File file = new File(artifact.toString());
        file.setExecutable(true, true);
```

Reply:
I had forgotten about that one. Yes, that one looks much simpler and more relevant to what we want. One minor nit - instead of `new File(artifact.toString())` , you can just do `artifact.toFile().setExecutable(true, true);`

Reply:
Done!